sway/tree/arrange.c

```diff
@@ -314,14 +314,9 @@ void arrange_output(struct sway_output *output) {
 	if (config->reloading) {
 		return;
 	}
-	struct wlr_box output_box;
-	wlr_output_layout_get_box(root->output_layout,
-		output->wlr_output, &output_box);
-	output->lx = output_box.x;
-	output->ly = output_box.y;
-	output->width = output_box.width;
-	output->height = output_box.height;
-
+	if (!output->wlr_output->enabled) {
```

Comment:
We want to add this check to `transaction.c` as well. Look at the function `arrange_root()` in there. But it looks like the outright removal of `wlr_output_layout_get_box()` is fine here because we do it anyway when we finalize the output config.

Reply:
So far my trick was to make `activated` in arrange_output dependent on `output->wlr_output->enabled`, so that the workspace trees get disabled. I think that works? It's a bit hard to see what happens on the screen before the output modesets...

We need all output specific nodes to be disabled while the output is so that not updating their position and size is okay, otherwise the unarranged content might overflow to neighbor outputs while our scene output geometry is zero-sized. :/

Reply:
I see, I missed that before.\